.github/workflows/buildAndTest.yml

```diff
@@ -266,21 +266,19 @@ jobs:
       # Run clang-tidy against only the changes. The 'clang-tidy-diff' script
       # does this if supplied with the diff.
       - name: clang-tidy
-        if: ${{ always() }}
         run: |
-          git diff -U0 $DIFF_COMMIT...HEAD | \
-            clang-tidy-diff -path build -p1 -fix -j$(nproc)
-          git clang-format -f $DIFF_COMMIT
-          git diff --ignore-submodules > clang-tidy.patch
-          if [ -s clang-tidy.patch ]; then
-            echo "Clang-tidy problems in the following files. " \
-              "See diff in the clang-tidy.patch artifact."
-            git diff --ignore-submodules --name-only
-            git checkout .
```

Comment:
IIRC we had `git checkout .` to put it back the way we found it. There may not be any steps after this which operate on the source code, but there could be eventually. I think it should be added back.

Reply:
Sounds good!

Reply:
Does anyone actually use the clang-tidy patch artifacts? I could get rid of it, while I'm here. I don't personally use it.  I'm happy to leave it in if it's useful (or if we're not sure).

Reply:
Leaving it in.